iModelCore/ECDb/ECDb/IntegrityChecker.cpp

```diff
@@ -994,15 +994,44 @@ DbResult IntegrityChecker::CheckClassIds(std::function<bool(Utf8CP, ECInstanceId
 				m_lastError = "failed to prepared ecsql for nav prop integrity check";
 				return BE_SQLITE_ERROR;
 			}
-			while((rc = stmt.Step()) == BE_SQLITE_ROW) {
-				if (!callback(
-					classCP->GetFullName(),
-					stmt.GetValueId<ECInstanceId>(0),
-					stmt.GetValueId<ECClassId>(1), "joined")) {
+			auto classIsValid = true;
+			while(stmt.Step() == BE_SQLITE_ROW)
+				{
+				if (!callback(classCP->GetFullName(), stmt.GetValueId<ECInstanceId>(0), stmt.GetValueId<ECClassId>(1), "joined"))
 					return BE_SQLITE_OK;
+				classIsValid = false;
 				}
-			}
-		}
+
+			if (classIsValid)
+				{
+				// Check missing child rows from BisCore:Element
+				ECSqlStatement getChildTablesStmt;
+				if (ECSqlStatus::Success != getChildTablesStmt.Prepare(m_conn, SqlPrintfString("select distinct ECClassId from %s", classCP->GetECSqlName().c_str()).GetUtf8CP()))
+					{
+					m_lastError = "Failed to prepare ecsql for classid integrity check";
+					return BE_SQLITE_ERROR;
+					}
+				while (getChildTablesStmt.Step() == BE_SQLITE_ROW)
+					{
+					if (const auto joinedClassCP = m_conn.Schemas().GetClass(getChildTablesStmt.GetValueId<ECClassId>(0)); joinedClassCP != nullptr)
+						{
+						const auto joinedClassId = joinedClassCP->GetId().ToHexStr();
+						ECSqlStatement getMissingRowsStmt;
+						if (ECSqlStatus::Success != getMissingRowsStmt.Prepare(m_conn, SqlPrintfString("select ECInstanceId from bis.Element where ECClassId=%s and ECInstanceId not in (select ECInstanceId from %s where ECClassId=%s)",
```

Comment:
Although it may seem correct, this approach will not solve the problem. Consider viewing the issue from a different angle. It concerns the existence of a row that should be present across multiple tables. I have written a query that, given the current instances in bis_Element, provides me with a group of classid that must have a corresponding row in a set of tables.  So you have to run like single query per permutation, which is like 7 in following IModel with over a million rows. Taking union of 7 will make it one query which will run really fast.

Following is the result

| Tables                                                             | GROUP_CONCAT(Id)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |   |   |   |
|--------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---|---|---|
| bis_Element, bis_DefinitionElement                                  | 94,105,108,109,110,117,121,129,139,140,152,169,172,190,193,196,214,237,255,342,346,349,363                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |   |   |   |
| bis_Element, bis_Document                                           | 146,228                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |   |   |   |
| bis_Element, bis_GeometricElement2d                                 | 147,226,249                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |   |   |   |
| bis_Element, bis_GeometricElement3d                                 | 252,264,270,274,1790,1962,2949,2988,3041,3188,3516,3616,4621,4842,4863,4898,5085,5179,5298,5299,5329,5537,5538,5539,6215,6216,6217,6262,6263,8117,8127,8128,8130                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |   |   |   |
| bis_Element, bis_GeometricElement3d, bis_GeometricElement3d_Overflow | 570,578,632,642,643,645,1677,1694,1797,1964,1985,1986,1987,1989,1991,1992,2010,4228,4229,4306,4308,4309,4320,4325,4331,4339,4340,4342,4344,4345,4356,4369,4373,4402,4403,4404,4411,4431,4442,4445,4447,4449,4450,4461,4463,4474,4500,4503,4506,4508,4518,4526,4527,4528,4532,4539,4540,4542,4545,4550,4552,4553,4555,4557,4559,4560,4561,4562,4563,4566,4622,4758,4763,4770,4807,4878,4887,4889,4901,4904,4905,4915,4919,4934,4949,4964,4975,4978,4979,4986,4996,5010,5024,5039,5042,5044,5055,5057,5060,5062,5066,5070,5080,5081,5091,5102,5108,5133,5136,5138,5139,5140,5143,5151,5171,5197,5202,5203,5206,5208,5215,5233,5235,5247,5248,5265,5271,5305,5308,5314,5342,5354,5368,5372,6137,6141,6145,6156,6161,6162,6165,6166,6168,6169,6170,6200 |   |   |   |
| bis_Element, bis_InformationPartitionElement                        | 136,145,178,183,192,211,242                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |   |   |   |
| bis_Element, bis_InformationReferenceElement                        | 200,244,267,279,281,300

Reply:
So in above table you can create a query like
```sql
SELECT 
    p.Id, j.ElementId, 
    o.ElementId 
FROM bis_Element p 
    LEFT JOIN bis_GeometricElement3d j ON j.ElementId=p.Id 
    LEFT JOIN bis_GeometricElement3d_Overflow o ON o.ElementId=.p.Id 
WHERE
    p.ECClassId IN( 570,578,632,642,643,64...) AND 
    (o.ElementId IS NULL OR j.ElementId IS NULL)
```

Reply:
also query to get that table
```sql
SELECT 
       [Tables], 
       GROUP_CONCAT ([Id])
FROM   (SELECT 
               [CL].[Id] [Id], 
               GROUP_CONCAT ([Tb].[Name]) [Tables]
        FROM   (SELECT DISTINCT [ECClassId] [Id]
                FROM   [bis_Element]) CL
               JOIN [ec_cache_ClassHasTables] [CT] ON [CT].[ClassId] = [CL].[Id]
               JOIN [ec_Table] [TB] ON [TB].[Id] = [CT].[TableId]
        GROUP  BY [CL].[Id])
GROUP  BY [Tables];
```

Reply:
in output you can show following table

|ECInstanceId|ECClassId|MissingRowInTables|
|--------------|-----------|------------------------|
|0x111           |  0x222      |  ["bis_GeometricElement3d", "bis_GeometericElement_Overflow"] |

Reply:
Resolving this convo as Affan has approved the changes.